src/docfx/lib/Range.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Docs.Build
+{
+    internal struct Range
+    {
+        public Range(int startLine, int startCharacter, int endLine = 0, int endCharacter = 0)
```

Comment:
>Range [](start = 15, length = 5)

since you created this constructor, why not make Range readonly?

`internal readonly struct Range
{
     public readonly int StartLine;
}`

Reply:
fixed